ddtrace/contrib/django/patch.py

```diff
@@ -345,11 +357,17 @@ def traced_get_response(django, pin, func, instance, args, kwargs):
         span._metrics[SPAN_MEASURED_KEY] = 1
 
         response = None
+
+        if _context.get_item("http.request.blocked", span=span):
+            func = HttpResponseForbidden
+            args = [appsec_utils._get_blocked_template(request_headers.get("Accept"))]
+
         try:
             response = func(*args, **kwargs)
             return response
         finally:
             # DEV: Always set these tags, this is where `span.resource` is set
+            _reset_contextvars()
```

Comment:
I didn't want to call the response object instantiation before `func` in case there were some side effects to it (now or in the future) but your code look clearer and they don't seem to be any so go with your suggestion.